frontend/src/scenes/notebooks/Nodes/NotebookNodeQuery.tsx

```diff
@@ -158,28 +158,31 @@ export const Settings = ({
 
 export const NotebookNodeQuery = createPostHogWidgetNode<NotebookNodeQueryAttributes>({
     nodeType: NotebookNodeType.Query,
-    title: async (attributes) => {
-        const query = attributes.query
-        let title = 'HogQL'
-        if (NodeKind.SavedInsightNode === query.kind) {
-            const response = await api.insights.loadInsight(query.shortId)
-            title = response.results[0].name?.length
-                ? response.results[0].name
-                : response.results[0].derived_name || 'Saved insight'
-        } else if (NodeKind.DataTableNode === query.kind) {
-            if (query.source.kind) {
-                title = query.source.kind.replace('Node', '').replace('Query', '')
-            } else {
-                title = 'Data exploration'
+    title: {
+        recompute: (attributes) => attributes.query.kind === NodeKind.SavedInsightNode,
+        value: async (attributes) => {
+            const query = attributes.query
+            let title = 'HogQL'
+            if (NodeKind.SavedInsightNode === query.kind) {
+                const response = await api.insights.loadInsight(query.shortId)
+                title = response.results[0].name?.length
+                    ? response.results[0].name
+                    : response.results[0].derived_name || 'Saved insight'
+            } else if (NodeKind.DataTableNode === query.kind) {
+                if (query.source.kind) {
+                    title = query.source.kind.replace('Node', '').replace('Query', '')
+                } else {
+                    title = 'Data exploration'
+                }
+            } else if (NodeKind.InsightVizNode === query.kind) {
+                if (query.source.kind) {
+                    title = query.source.kind.replace('Node', '').replace('Query', '')
+                } else {
+                    title = 'Insight'
+                }
             }
-        } else if (NodeKind.InsightVizNode === query.kind) {
-            if (query.source.kind) {
-                title = query.source.kind.replace('Node', '').replace('Query', '')
-            } else {
-                title = 'Insight'
-            }
-        }
-        return Promise.resolve(title)
+            return Promise.resolve(title)
+        },
```

Comment:
I might be wrong here but I have some weird feelings about the overall solution...
In particular it feels hacky that we are calling an api request here - especially given the underlying query will load the insight anyways 🤔 Now every time a notebook loads we call the insights api - not good.

Could we not remove all of this, keep the title a simple string, and then allow the component to call `updateAttributes` if it determines the title has changed? 

I feel like we could go:
1. We don't set the title attribute by default, we just fallback to whatever is given by the `createPostHogWidgetNode` options
2. The component can call `setTitle` - this allows us to add a useEffect or whatever else _in_ the component and handle lifecycle things, optimising for avoiding overcalling the api.
3. If the component has set a title that is what will be used, until it says otherwise

This way we can remove the increasingly complex logic here whilst still getting all we need out of it, I think?